### PR TITLE
Add tests + fix bug introduced in #5

### PR DESCRIPTION
**Pull request by jesstelford**
I also ran `prettier` over the code + tests 👍 

```
❯ yarn test
yarn run v1.12.3
warning package.json: No license field
$ jest
 PASS  test/index.test.js
  [with prefix my_prefix]
    ✓ prefixURL relative URLs (163ms)
    ✓ prefixURL absolute URLs (21ms)
    ✓ prefixURL object URLs (15ms)
    ✓ Image (29ms)
    ✓ Link (22ms)
  [with prefix a-prefix/]
    ✓ prefixURL relative URLs (13ms)
    ✓ prefixURL absolute URLs (14ms)
    ✓ prefixURL object URLs (15ms)
    ✓ Image (15ms)
    ✓ Link (14ms)
  [with prefix a/path]
    ✓ prefixURL relative URLs (14ms)
    ✓ prefixURL absolute URLs (14ms)
    ✓ prefixURL object URLs (12ms)
    ✓ Image (11ms)
    ✓ Link (13ms)
  [with prefix some-prefix////]
    ✓ prefixURL relative URLs (11ms)
    ✓ prefixURL absolute URLs (13ms)
    ✓ prefixURL object URLs (10ms)
    ✓ Image (10ms)
    ✓ Link (12ms)

----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |      100 |      100 |      100 |      100 |                   |
 index.js |      100 |      100 |      100 |      100 |                   |
----------|----------|----------|----------|----------|-------------------|
Test Suites: 1 passed, 1 total
Tests:       20 passed, 20 total
Snapshots:   16 passed, 16 total
Time:        2.418s
Ran all test suites.
✨  Done in 3.94s.
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,15 +4,17 @@ import getConfig from 'next/config';
...
+  publicRuntimeConfig.assetPrefix.replace(/\/+$/, '') +
+  '/' +
+  (path || '').replace(/^\/+/, '');
 
 export const prefixURL = url => {
   if (typeof url === 'object') {
-    return { ...url, pathname: prefixPath(url.pathname) };
-  } else {
-    return /^(?:[a-z][a-z0-9+.-]*:|\/\/)/i.test(url) ? url : prefixPath(url);
+    return Object.assign({}, url, { pathname: prefixPath(url.pathname) });
```
Comment (jesstelford) [inline src/index.js:14]: NOTE: I changed the object spread to an `Object.assign()` to reduce the babelified gzipped size by 200 bytes 👍

Comment (hipstersmoothie) [reply src/index.js:14]: you're awesome 🥇

**Comment by hipstersmoothie** [comment]
@jesstelford I turned builds on for forked PRs. Something seems to be up with the tests

**Comment by hipstersmoothie** [comment]
@jesstelford I think all you need to do is update jest 

https://github.com/facebook/jest/issues/8069

**Comment by jesstelford** [comment]
`jest` upgraded 👍

**Comment by jesstelford** [comment]
🚀

**Pull request merged**